daffodil-runtime1/src/main/scala/org/apache/daffodil/runtime1/infoset/JsonInfosetOutputter.scala

```diff
@@ -29,12 +29,15 @@ import org.apache.daffodil.runtime1.api.InfosetSimpleElement
 
 import com.fasterxml.jackson.core.io.JsonStringEncoder
 
-class JsonInfosetOutputter private (writer: java.io.Writer, pretty: Boolean)
+class JsonInfosetOutputter private (writer: java.io.BufferedWriter, pretty: Boolean)
   extends InfosetOutputter
   with Indentable {
 
   def this(os: java.io.OutputStream, pretty: Boolean) = {
-    this(new java.io.OutputStreamWriter(os, StandardCharsets.UTF_8), pretty)
+    this(
+      new java.io.BufferedWriter(new java.io.OutputStreamWriter(os, StandardCharsets.UTF_8)),
```

Comment:
Add a comment here that this buffered writer provides significant performance improvement. 

My first thought was that you should only add a Buffered writer if os argument is non-buffered, so you should conditionally check if the os output stream is already buffered. 

But we need a writer wrapped around the output stream anyway. And even if the output stream is buffered, use of a buffered writer may still  speed things up.

Reply:
> Add a comment here that this buffered writer provides significant performance improvement.

WIll do.

> But we need a writer wrapped around the output stream anyway.

Correct. When I was looking at the profiler, every write required spinning up a stream encoder, which has a lot of overhead. It's much more efficient to encode a bunch of characters at once. So we need a `BufferedWriter` regardless of what the `os` parameter is.

I just tested what happens if the `os` parameter is already a `BufferedOutputStream` (i.e. essentially double buffering), and it doesn't make a noticeable difference in parse time. So it's unnecessary to pass in a `BufferedOutuptStream`, but it doesn't hurt.